level/doc.go

```diff
@@ -19,4 +19,13 @@
 // NewFilter allows precise control over what happens when a log event is
 // emitted without a level key, or if a squelched level is used. Check the
 // Option functions for details.
+//
+// It's also possible configure the filter passwind a string value
+// (for instance read from flag, environment variable, .....)
+//
+//    var logger log.Logger
+//    logger = log.NewLogfmtLogger(os.Stderr)
+//    logger = level.NewFilter(logger, level.AllowInfo(), level.Parse(os.Getenv("LOG_LEVEL"))) // <--
+//    logger = log.With(logger, "ts", log.DefaultTimestampUTC)
```

Comment:
Hmm, it seems like level should be established before the filter is constructed. (Also, this is a minor nit, but nothing should read environment variables directly like this, either.)

So the construction site should look something like

```go
var logger log.Logger
logger = log.NewLogfmtLogger(os.Stderr)
logger = level.NewFilter(logger, allow)
logger = log.With(logger, "ts", log.DefaultTimestampUTC)
```

Working backwards, I'd expect something like

```go
var (
    ...
    logLevel = fs.String("log-level", "info", "debug, info, warn, error")
    ...
)

allow, err := level.Parse(*logLevel)
if err != nil {
    return fmt.Errorf("log level: %w", err)
}
```